src/segment_git.go

```diff
@@ -131,12 +146,44 @@ func (g *git) string() string {
 	if g.props.getBool(StatusColorsEnabled, false) {
 		g.SetStatusColor()
 	}
+	enableHyperlink := g.props.getBool(EnableHyperlink, false)
 	buffer := new(bytes.Buffer)
+
+	// url
+	upstream := replaceAllString("/.*", g.repo.upstream, "")
+	g.repo.url = g.getGitCommandOutput("remote", "get-url", upstream)
+
 	// branchName
 	if g.repo.upstream != "" && g.props.getBool(DisplayUpstreamIcon, false) {
-		fmt.Fprintf(buffer, "%s", g.getUpstreamSymbol())
+		fmt.Fprintf(buffer, "%s", g.getUpstreamSymbol(g.repo.url))
 	}
-	fmt.Fprintf(buffer, "%s", g.repo.HEAD)
+	// default name
+	repoHEAD := g.repo.PRETTYHEAD
+	if g.repo.upstream != "" && enableHyperlink {
+		// parse url to and strips extra info
+		baseURL, err := url.Parse(g.repo.url)
+		if err == nil {
+			var urlTemplate string
+			// check for a match in the providers map
+			for i, val := range gitProvidersMap {
+				if strings.Contains(baseURL.Host, i) {
+					urlTemplate = val
+					break
+				}
+			}
+			// construct url(remove .git from url(github))
+			// removing extra data could be done using a regex stored in the map
+			repoURL := fmt.Sprintf("%s://%s%s", baseURL.Scheme, baseURL.Host, strings.ReplaceAll(baseURL.Path, ".git", ""))
+			if urlTemplate != "" {
+				// if provider found, open the branch
+				repoHEAD = fmt.Sprintf("]8;;%s\\%s]8;;\\", fmt.Sprintf(urlTemplate, repoURL, g.repo.HEAD), g.repo.PRETTYHEAD)
+			} else {
```

Comment:
once this logic is in a separate function, the else can be removed when you return a string and write this in the function above